### PR TITLE
refactor: update TaskMetadata

### LINKED ISSUE #1054: Update TaskMetadata
(not intended before the MMTEB release)

From #1051 and general review it seems like a reasonable update the TaskMetadata object as follows:

```diff
    metadata = TaskMetadata(
        name="BUCC.v2",
        dataset={
            "path": "mteb/bucc-bitext-mining",
            "revision": "1739dc11ffe9b7bfccd7f3d585aeb4c544fc6677",
        },
        description="BUCC bitext mining dataset",
        reference="https://comparable.limsi.fr/bucc2018/bucc2018-task.html",
        type="BitextMining",
        category="s2s", # can stay (e.g. i2t - image2text), but we could rename?
        eval_splits=_SPLITS,
        eval_langs=_LANGUAGES, # keep, but knowing that it can be empty for image-only tasks
        main_score="f1",
        date=("2017-01-01", "2018-12-31"),
-        form=["written"], # merge into domains
-        domains=[],
+        domains=["Written"],
+      modalities=["Text"], # new can be mulitple, includes ["text", "image"] 
        task_subtypes=[],
        license="unknown",
-        socioeconomic_status="mixed", # mostly confusing
        annotations_creators="human-annotated",
        dialect=[], # probably keep, but doesn't make as much sense for images
-       text_creation="human-translated",
+       sample_creation="human-translated",
        bibtex_citation="""{redacted for readability}""",
+        descriptive_statistics = {"n_samples": {"test": 641684}, "avg_character_length": {"test": 101.3}}
-        n_samples={"test": 641684},
-        avg_character_length={"test": 101.3},

```

### PR DESCRIPTION
**Pull request by MartinBernstorff**
@KennethEnevoldsen In case of merge-conflicts, we can just re-use the `.grit` script and re-apply using:

```shell
grit apply _add_stats.grit tests
grit apply _add_stats.grit mteb
```

Fixes #1054.

Likely discovered a bug in GritQL, report [here](https://github.com/getgrit/gritql/issues/416).

### CODE REVIEW

File: docs/adding_a_dataset.md
Comment (KennethEnevoldsen) [inline docs/adding_a_dataset.md:1]: the examples in this file need to be updated to the newest version

File: mteb/tasks/BitextMining/kat/TbilisiCityHallBitextMining.py [hunk 1]
```diff
@@ -28,22 +28,23 @@ class TbilisiCityHallBitextMining(AbsTaskBitextMining, MultilingualTask):
...
         date=("2024-05-02", "2024-05-03"),
-        form=["written"],
         task_subtypes=[],
         license="Not specified",
-        socioeconomic_status="mixed",
         annotations_creators="derived",
         dialect=[],
         bibtex_citation="",
-        avg_character_length={_EVAL_SPLIT: 78},
+        stats={
```
Comment (KennethEnevoldsen) [inline mteb/tasks/BitextMining/kat/TbilisiCityHallBitextMining.py:44]: would probably call it descriptive_statistics

File: tests/test_retrieval_abstask.py [hunk 2]
```diff
@@ -7,3 +7,8 @@
 @pytest.mark.parametrize("task", [NFCorpus()])
 def test_abstask_calculate_metadata_metrics(task: AbsTaskRetrieval):
     task.calculate_metadata_metrics()
+
+
+if __name__ == "__main__":
+    pass
+    pass
```
Comment (KennethEnevoldsen) [inline tests/test_retrieval_abstask.py:14]:
```suggestion
```

**Review by KennethEnevoldsen — COMMENTED** [review]
This looks great, thanks Martin! Also seems like it didn't take too long.

Yea we will def. have a merge conflict with #1070. 

However, it seems like we can mostly resolve that by branching from that and re-applying the change (which is great!)

Will just add @isaac-chung and @gowitheflow-1998 as reviewers here as well.

**Comment by MartinBernstorff** [comment]
> This looks great, thanks Martin! Also seems like it didn't take too long.
> 
> Yea we will def. have a merge conflict with #1070.
> 
> However, it seems like we can mostly resolve that by branching from that and re-applying the change (which is great!)
> 
> Will just add @isaac-chung and @gowitheflow-1998 as reviewers here as well.

Haha, yeah, if I were proficient with GritQL, it would definitely be fast. Took me a couple of hours, but much faster next time.

**Review by gowitheflow-1998 — APPROVED** [review]
looks great! Thanks for updating these. Adding `modalities` and changing from `text_creation` to `sample_creation` definitely makes it easier for merging meta data from the image branch and other modalities in the future. Agree with `stats` -> `descriptive_statistics` otherwise not much to add on my end.

**Comment by MartinBernstorff** [comment]
@KennethEnevoldsen I've added a few points (6). Feel free to lower the amount, and then it's ready to merge 👍

**Pull request merged**